KomaMRIBase/src/datatypes/Phantom.jl

```diff
@@ -612,5 +539,62 @@ function check_phantom_arguments(nd, ss, us)
             ssy = ss[2]
         end
     end
+    
     return ssx, ssy, ssz, usx, usy, usz
 end
+
+"""
+    T1, T2, T2s, ρ, Δw = default_brain_tissue_properties(labels, tissue_properties = nothing)
+
+This function returns the default brain tissue properties using a labels identifier Matrix
+# Arguments
+- `labels` : (`::Matrix`) the labels identifier matrix of the phantom
+- `tissue_properties` : (`::Dict`, `=Dict()`) phantom tissue properties in ms and Hz considering the available tissues
+
+# Returns
+- `T1, T2, T2s, ρ, Δw`: (`::Matrix`) matrices of the same size of labels with the tissues properties information
+
+# Examples
+```julia-repl
+julia> T1, T2, T2s, ρ, Δw = default_brain_tissue_properties(labels, tissue_properties)
+
+julia> T1, T2, T2s, ρ, Δw = default_brain_tissue_properties(labels)
+```
+"""
+function default_brain_tissue_properties(labels, tissue_properties = Dict())
+    # Load default tissue properties
+    default_properties = Dict(
+        # T1, T2, T2*, ρ, Δw
+        "CSF" => [2.569, 0.329, 0.058, 1, 0],
+        "GM" => [0.833, 0.083, 0.069, 0.86, 0],
+        "WM" => [0.500, 0.070, 0.061, 0.77, 0],
+        "FAT1" => [0.350, 0.070, 0.058, 1, -3.84], #-220 Hz
+        "MUSCLE" => [0.900, 0.047, 0.030, 1, 0],
+        "SKIN/MUSCLE" => [0.569, 0.329, 0.058, 1, 0],
+        "SKULL" => [0, 0, 0, 0, 0],
+        "VESSELS" => [0, 0, 0, 0, 0],
+        "FAT2" => [0.500, 0.070, 0.061, 0.77, -3.84], #-220 Hz
+        "DURA" => [2.569, 0.329, 0.058, 1, 0],
+        "MARROW" => [0.500, 0.070, 0.061, 0.77, 0])
+    
+    tissue_properties = merge(default_properties, tissue_properties)
+    flat_labels = reshape(labels, :)
+    properties = zeros(length(flat_labels)..., 5)
+    for i=1:5
+        properties[:, i] = 
+        (flat_labels .== 23) * tissue_properties["CSF"][i] .+ #CSF
+        (flat_labels .== 46) * tissue_properties["GM"][i] .+ #GM
+        (flat_labels .== 70) * tissue_properties["WM"][i] .+ #WM
+        (flat_labels .== 93) * tissue_properties["FAT1"][i] .+ #FAT1
+        (flat_labels .== 116) * tissue_properties["MUSCLE"][i] .+ #MUSCLE
+        (flat_labels .== 139) * tissue_properties["SKIN/MUSCLE"][i] .+ #SKIN/MUSCLE
+        (flat_labels .== 162) * tissue_properties["SKULL"][i] .+ #SKULL
+        (flat_labels .== 185) * tissue_properties["VESSELS"][i] .+ #VESSELS
+        (flat_labels .== 209) * tissue_properties["FAT2"][i] .+ #FAT2
+        (flat_labels .== 232) * tissue_properties["DURA"][i] .+ #DURA
+        (flat_labels .== 255) * tissue_properties["MARROW"][i] #MARROW
+    end
+    properties = reshape(properties, (size(labels)..., 5))
+
+    return selectdim(properties, ndims(properties), 1), selectdim(properties, ndims(properties), 2), selectdim(properties, ndims(properties), 3), selectdim(properties, ndims(properties), 4), selectdim(properties, ndims(properties), 5)
```

Comment:
This is overly complicated.

```julia
return (properties[:, i] for i in 1:Nproperties)
```

Reply:
I tried that approach. However, with that idea, a flat vector will be returned and a reshape should be done on the `brain_phantom_2D` function. Using `selectdim` generalizes for either 2D or 3D implementations and no reshape is required twice on the 2D and 3D functions.